metricflow/specs/where_filter_dimension.py

```diff
@@ -0,0 +1,64 @@
+from typing import Sequence
+from metricflow.specs.column_assoc import ColumnAssociationResolver
+from metricflow.specs.query_interface import QueryInterfaceDimension
+from dbt_semantic_interfaces.naming.dundered import DunderedNameFormatter
+
+from typing import Sequence, Type
+
+from dbt_semantic_interfaces.call_parameter_sets import (
+    DimensionCallParameterSet,
+)
+from dbt_semantic_interfaces.naming.dundered import DunderedNameFormatter
+from dbt_semantic_interfaces.references import (
+    DimensionReference,
+    EntityReference,
+)
+
+from metricflow.specs.query_interface import (
+    QueryInterfaceDimension,
+)
+from metricflow.specs.specs import DimensionSpec
+from dbt_semantic_interfaces.call_parameter_sets import FilterCallParameterSets
+
+
+def get_where_filter_dimension_cls(
+    call_parameter_sets: FilterCallParameterSets,
+    dimension_specs: DimensionSpec,
+    column_association_resolver: ColumnAssociationResolver,
+):
+    class WhereFilterDimension(QueryInterfaceDimension):
```

Comment:
Anonymous class patterns of this sort are not common in Python. It's much more typical - and more efficient in the runtime - to define a class and then initialize it inside a function. Returning a closure for the purposes of having a callable available to jinja template processing is fairly straightforward by comparison.

I might be missing something, but this is a fairly specific class specification. I think we'll find it's most natural to have these classes defined and importable, and then to wrap each one in a closure where necessary, instead of making each class initializer the equivalent of an anonymous function accessed through a separate method like this.